flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/hybrid/HybridSourceSplitEnumeratorTest.java

```diff
@@ -177,6 +177,26 @@ public void testRestoreEnumerator() throws Exception {
                 .hasSize(1);
     }
 
+    @Test
+    public void testRestoreEnumeratorAfterFirstSourceWithoutRestoredSplits() throws Exception {
+        setupEnumeratorAndTriggerSourceSwitch();
+        HybridSourceEnumeratorState enumeratorState = enumerator.snapshotState(0);
+        MockSplitEnumerator underlyingEnumerator = getCurrentEnumerator(enumerator);
+        assertThat(
+                        (List<MockSourceSplit>)
+                                Whitebox.getInternalState(underlyingEnumerator, "splits"))
+                .hasSize(0);
+        enumerator =
+                (HybridSourceSplitEnumerator) source.restoreEnumerator(context, enumeratorState);
+        enumerator.start();
+        enumerator.handleSourceEvent(SUBTASK0, new SourceReaderFinishedEvent(-1));
```

Comment:
```suggestion
        // subtask starts at -1 since it has no splits after restore
        enumerator.handleSourceEvent(SUBTASK0, new SourceReaderFinishedEvent(-1));
```

Reply:
done.